tools/trace_analyzer_tool_imp.h

```diff
@@ -0,0 +1,306 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+#ifndef ROCKSDB_LITE
+
+#include "rocksdb/trace_analyzer_tool.h"
+
+#include <stdio.h>
+#include <fstream>
+#include <list>
+#include <map>
+#include <queue>
+#include <set>
+#include <utility>
+#include <vector>
+
+#include "rocksdb/env.h"
+#include "rocksdb/write_batch.h"
+#include "util/trace_replay.h"
+
+namespace rocksdb {
+
+class DBImpl;
+class WriteBatch;
+class AnalyzerOptions;
+class TraceWriteHandler;
+
+const int taTypeNum = 7;
+
+enum TraceOperationType : uint32_t {
+  taGet = 0,
+  taPut = 1,
+  taDelete = 2,
+  taSingleDelete = 3,
+  taRangeDelete = 4,
+  taMerge = 5,
+  taIter = 6
+};
+
+struct TraceUnit {
+  uint32_t type;
+  std::string key;
+  size_t value_size;
+  uint64_t ts;
+  uint32_t cf_id;
```

Comment:
I have added to "CMakeLists.txt" and "src.mk" to build it. However, since the trace_replay is not in the master yet, the trace_analyzer cannot be build. So I removed the change to that two files before I create the pull requests. Later on, I will add the change back.